core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -843,6 +669,274 @@ private void syncLocalToRemote() {
         logicalT));
   }
 
+  /**
+   * Synchronizes the remote state of every requested document to be synchronized with the local
+   * state of said documents. Utilizes change streams to get "recent" updates to documents of
+   * interest. Documents that are being synchronized from the first time will be fetched via a
+   * full document lookup. Any conflicts that occur will be resolved locally and later relayed
+   * remotely on a subsequent iteration of {@link DataSynchronizer#doSyncPass()}.
+   */
+  private void syncRemoteToLocal() {
+    logger.info(String.format(
+        Locale.US,
+        "t='%d': syncRemoteToLocal START",
+        logicalT));
+
+    for (final NamespaceSynchronizationConfig nsConfig : syncConfig) {
+      final MongoCollection<BsonDocument> localColl = getLocalCollection(nsConfig.getNamespace());
+      final Map<BsonValue, ChangeEvent<BsonDocument>> remoteChangeEvents =
+          instanceChangeStreamListener.getEventsForNamespace(nsConfig.getNamespace());
+
+      final Set<BsonValue> unseenIds = nsConfig.getSynchronizedDocumentIds();
+      final Set<BsonDocument> staleDocuments = getStaleDocuments(nsConfig);
+      final Set<BsonValue> staleIds = getStaleDocumentIds(staleDocuments);
+      final Map<BsonValue, BsonDocument> staleDocumentMap = new HashMap<>();
+      for (final BsonDocument staleDocument : staleDocuments) {
+        staleDocumentMap.put(staleDocument.get("_id"), staleDocument);
+      }
+
+      for (final Map.Entry<BsonValue, ChangeEvent<BsonDocument>> eventEntry :
+          remoteChangeEvents.entrySet()) {
+        logger.info(String.format(
+            Locale.US,
+            "t='%d': syncRemoteToLocal consuming event of type: %s",
+            logicalT,
+            eventEntry.getValue().getOperationType()));
+
+        final CoreDocumentSynchronizationConfig docConfig =
+            nsConfig.getSynchronizedDocument(eventEntry.getKey().asDocument().get("_id"));
+
+        if (docConfig == null) {
+          // Not interested in this event.
+          continue;
+        }
+
+        unseenIds.remove(docConfig.getDocumentId());
+        staleIds.remove(docConfig.getDocumentId());
+        syncRemoteChangeEventToLocal(nsConfig, docConfig, eventEntry.getValue(), localColl);
+      }
+
+      for (final BsonValue docId : unseenIds) {
+        final CoreDocumentSynchronizationConfig docConfig =
+            nsConfig.getSynchronizedDocument(docId);
+        if (docConfig == null) {
+          // means we aren't actually synchronizing on this remote doc
+          continue;
+        }
+
+        if (nsConfig.isStale() && staleIds.contains(docId)) {
+          if (docConfig.hasCommittedVersion(DocumentVersionInfo
```

Comment:
This check should be covered in the sync remote to local.

Reply:
👍